optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/InnerConstraintFactory.java

```diff
@@ -73,6 +78,34 @@
         }
     }
 
+    public <A> void assertValidFromClass(Class<A> fromClass) {
```

Comment:
It seems this could be `protected` to indicate it's only meant to be called by subclasses.

Reply:
Unfortunately not, I also need to call this from the streams.